lib/segment/src/vector_storage/chunked_utils.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use memmap2::MmapMut;
+
+use crate::common::mmap_ops::{
+    create_and_ensure_length, open_write_mmap, transmute_from_u8_to_mut_slice_shared,
+};
+use crate::data_types::vectors::VectorElementType;
+use crate::entry::entry_point::{OperationError, OperationResult};
+
+const MMAP_CHUNKS_PATTERN_START: &str = "chunk_";
+const MMAP_CHUNKS_PATTERN_END: &str = ".mmap";
+
+pub struct MmapChunk {
+    /// Memory mapped file for chunk data.
+    ///
+    /// This should never be accessed directly, because it shares a mutable reference with
+    /// [`data`]. Use that instead. The sole purpose of this is to keep ownership of
+    /// the mmap, and to properly clean it up when this struct is dropped.
+    pub _mmap: Arc<MmapMut>,
+    /// A convenient `&mut [VectorElementType]` view into the memory map file.
+    ///
+    /// This has the same lifetime as this struct, a borrow must never be leased out for longer.
+    pub data: &'static mut [VectorElementType],
```

Comment:
The lifetime of `data` isn't `'static`, you're only able to cast this to a `'static` lifetime because of unbounded lifetimes. Given that `data` is `pub`, you should probably try to avoid this. Even though the way you're currently using `data` seems fine, this might still be abused by someone in the future with potentially bad consequences.

Reply:
Totally make sense, will make it private instead.